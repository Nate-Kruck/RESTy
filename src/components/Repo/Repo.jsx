import React from 'react';
import PropTypes from 'prop-types';

function Repo({ repo }) {
  return (
    <div>
      <li>
        <a href={repo.url} target="_blank" rel="noreferrer">
          <h1 data-testid="repo">{repo.name}</h1>
        </a>
      </li>
      <p>{repo.login}</p>
    </div>
  );
}

Repo.propTypes = {
  repo: PropTypes.shape({
    url: PropTypes.string,
    name: PropTypes.string,
    login: PropTypes.string
  }).isRequired
};

export default Repo;
