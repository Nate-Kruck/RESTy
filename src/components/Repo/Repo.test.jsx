import React from 'react';
import {
  render,
  cleanup
} from '@testing-library/react';
import Repo from '../Repo/Repo';

describe('Repo Testing', () => {
  afterEach(() => cleanup());

  it('renders a Repo', () => {
    const { asFragment } = render(<Repo
      url="blank"
      method="GET"
      body="blank"
      onChange={() => {}}
      onSubmit={() => {}}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
