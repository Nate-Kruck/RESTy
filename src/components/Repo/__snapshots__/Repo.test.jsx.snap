// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Testing renders a Repo 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="url"
      >
        URL
        <input
          name="url"
          type="text"
          value="blank"
        />
      </label>
      <label
        for="method"
      >
        HTTP Request method
        <select
          name="method"
        >
          <option
            value="GET"
          >
            GET
          </option>
          <option
            value="POST"
          >
            POST
          </option>
          <option
            value="PUT"
          >
            PUT
          </option>
          <option
            value="PATCH"
          >
            PATCH
          </option>
          <option
            value="DELETE"
          >
            DELETE
          </option>
        </select>
      </label>
      <p>
        JSON data
      </p>
      <textarea
        name="body"
      >
        blank
      </textarea>
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
